core/src/main/java/org/elasticsearch/action/support/tasks/TransportTasksAction.java

```diff
@@ -124,6 +126,22 @@ private NodeTasksResponse nodeOperation(NodeTaskRequest nodeTaskRequest) {
         return clusterState.nodes().resolveNodesIds(request.nodesIds());
     }
 
+    protected void processTasks(TasksRequest request, Consumer<OperationTask> operation) {
+        if (request.taskId() != BaseTasksRequest.ALL_TASKS) {
+            // we are only checking one task, we can optimize it
+            Task task = taskManager.getTask(request.taskId());
+            if (task != null && request.match(task)) {
```

Comment:
The task id is specified by a user and by the time it reaches this node the task with the given id might be already gone or the user might have specified a wrong id to start with and this task might have never existed. In both cases taskManager would return null. I was thinking about throwing a TaskNotFoundException or something like this in this case, but at the end decided to just silently ignore the request for now. Not sure what would be a good user experience here.